src/tensors/cpu/intgemm_interface.h

```diff
@@ -0,0 +1,391 @@
+#pragma once
+
+#include "graph/node.h"
+#include "graph/node_operators_unary.h"
+#include "integer_common.h"
+
+namespace marian {
+
+namespace cpu {
+namespace integer {
+
+template<Type vtype>
+struct PrepareANodeOp : public NaryNodeOp {
```

Comment:
Please add class-level comments to every new node type describing what they do and what their purpose is.